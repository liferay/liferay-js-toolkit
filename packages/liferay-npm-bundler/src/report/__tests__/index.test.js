import PluginLogger from 'liferay-npm-build-tools-common/lib/plugin-logger';
import {Report} from '../index';

let report;

beforeEach(() => {
	report = new Report();

	// Hack to make tests repeatable
	report._executionDate = new Date(0);
});

describe('when describing the run', () => {
	afterEach(() => {
		expect(report).toMatchSnapshot();
	});

	it('correctly stores executionTime', () => {
		report.executionTime([1, 2]);
	});

	it('correctly stores warnings', () => {
		report.warn('warn 1');
		report.warn('warn 2');
	});

	it('correctly stores unique warnings', () => {
		report.warn('warn', {unique: true});
		report.warn('warn', {unique: true});
	});

	it('correctly stores versions info', () => {
		report.versionsInfo({
			'liferay-npm-bundler': '1.4.2',
			'liferay-npm-bundler-plugin-inject-angular': '1.4.2',
		});
	});

	it('correctly stores linked dependencies', () => {
		report.linkedDependency('a-package', 'file:../a-package', '1.1.0');
	});

	it('correctly stores dependencies', () => {
		report.dependencies([
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			{id: 'package-2@2.0.0', name: 'package-2', version: '2.0.0'},
		]);
	});

	it('correctly removes stale linked dependencies', () => {
		report.linkedDependency('a-package', 'file:../a-package', '1.1.0');
		report.linkedDependency('package-1', 'file:../package-1', '1.0.0');
		report.dependencies([
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			{id: 'package-2@2.0.0', name: 'package-2', version: '2.0.0'},
		]);
	});

	it('correctly stores package copies', () => {
		report.packageCopy(
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			['a.js', 'b.js', 'c.js'],
			['a.js', 'c.js'],
			['b.*']
		);
	});

	it('correctly stores root package bundler plugin runs', () => {
		report.rootPackageProcessBundlerPlugin(
			{
				name: 'a-plugin',
				config: {cfgval1: 1, cfgval2: 2},
				run: () => '',
			},
			new PluginLogger()
		);
	});

	it('correctly stores bundler plugin runs', () => {
		report.packageProcessBundlerPlugin(
			'pre',
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			{
				name: 'a-plugin',
				config: {cfgval1: 1, cfgval2: 2},
				run: () => '',
			},
			new PluginLogger()
		);
	});

	it('correctly stores Babel config', () => {
		report.packageProcessBabelConfig(
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			{cfgval1: 1, cfgval2: 2}
		);
	});

	it('correctly stores Babel file run', () => {
		report.packageProcessBabelRun(
			{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
			'a.js',
			new PluginLogger()
		);
	});
});

it('correctly dumps HTML report', () => {
	// The goal of this test is to detect unwanted changes in HTML. If you make
	// changes to the HTML on purpose, just check it visually and update the
	// snapshot with Jest's -u switch.

	report.executionTime([1, 2]);
	report.warn('warn 1');
	report.warn('warn 2');
	report.warn('warn', {unique: true});
	report.warn('warn', {unique: true});
	report.versionsInfo({
		'liferay-npm-bundler': '1.4.2',
		'liferay-npm-bundler-plugin-inject-angular': '1.4.2',
	});
	report.linkedDependency('a-package', 'file:../a-package', '1.1.0');
	report.linkedDependency('package-1', 'file:../package-1', '1.0.0');
	report.dependencies([
		{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
		{id: 'package-2@2.0.0', name: 'package-2', version: '2.0.0'},
	]);
	report.packageCopy(
		{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
		['a.js', 'b.js', 'c.js'],
		['a.js', 'c.js'],
		['b.*']
	);
	report.rootPackageProcessBundlerPlugin(
		{
			name: 'a-plugin',
			config: {cfgval1: 1, cfgval2: 2},
			run: () => '',
		},
		new PluginLogger()
	);
	report.packageProcessBundlerPlugin(
		'pre',
		{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
		{
			name: 'a-plugin',
			config: {cfgval1: 1, cfgval2: 2},
			run: () => '',
		},
		new PluginLogger()
	);
	report.packageProcessBabelConfig(
		{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
		{cfgval1: 1, cfgval2: 2}
	);
	report.packageProcessBabelRun(
		{id: 'package-1@1.0.0', name: 'package-1', version: '1.0.0'},
		'a.js',
		new PluginLogger()
	);

	expect(report.toHtml()).toMatchSnapshot();
});
