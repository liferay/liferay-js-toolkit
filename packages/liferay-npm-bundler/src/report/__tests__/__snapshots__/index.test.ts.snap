// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly dumps HTML report 1`] = `
"
		<!DOCTYPE html>
		<html>
			<head>
				<meta charset=\\"utf-8\\" />
				<title>Report of liferay-npm-bundler execution</title>
				<style>
					body, h1, h2, h3, p, li {
						font-family: sans-serif;
					}
					
					body, p, th, td, li {
						font-size: 10pt;
					}
					
					h1 {
						font-size: 16pt;
						margin: 1em 0 .5em 0;
					}
					
					h2 {
						font-size: 13pt;
						margin: 1em 0 .5em 0;
					}
					
					h3 {
						font-size: 11pt;
						margin: 1em 0 .5em 0;
					}
					
					table {
						margin: 0 0 .5em 0;
					}
					
					tr:nth-child(odd) {
						background-color: #F6F6F6;
					}
					
					th {
						background-color: #F0F0F0;
						text-align: left;
					}
					
					th, td {
						padding: .1em;
						vertical-align: top;
					}
					
					td.debug, td.info, td.warn, td.error {
						border-radius: 4px;
						color: white;
						padding: 0 2px;
						text-align: center;
						vertical-align: middle;
						width: 1px;
						white-space: nowrap;
					}

					td.debug {
						background: gray;
					}

					td.info {
						background: green;
					}

					td.warn {
						background: orange;
					}

					td.error {
						background: red;
					}
					
					td.source {
						color: grey;
						white-space: nowrap;
					}

					ul {
						padding-left: 1em;
						margin: 0 0 .5em 0;
					}
					
					p {
						margin: 0 0 .5em 0;
					}

					a {
						text-decoration: none;
						color: #055;
					}

					#log-level-selector {
						position: fixed;
						top: 1em;
						right: 1em;
						background-color: #eee;
						padding: .3em;
						border-radius: 4px;
						font-size: 8pt;
						border: 1px solid #ccc;
					}

					#log-level-selector select {
						font-size: 8pt;
					}

					.configuration {
						display: inline-block;
						margin-bottom: .5em;
					}

					.configuration > div {
						background-color: #f0f0f0;
						cursor: pointer;
						border-radius: 4px;
						padding: 2px;
						display: inline;
					}

					.configuration > div:after {
						content: \\"ðŸ‘€\\";
						padding: 0 .5em;
					}

					.configuration > pre {
						font-size: 8pt;
						display: none;
					}

					.configuration:hover > pre {
						display: block;
					}
				</style>
				<script id=\\"report\\" type=\\"application/json\\">
					{\\"_executionDate\\":\\"1970-01-01T00:00:00.000Z\\",\\"_executionTime\\":[1,2],\\"_versionsInfo\\":{\\"liferay-npm-bundler\\":{\\"version\\":\\"1.4.2\\",\\"path\\":\\"path/to/bundler\\"},\\"liferay-npm-bundler-plugin-inject-angular\\":{\\"version\\":\\"1.4.2\\",\\"path\\":\\"path/to/plugin\\"}},\\"_rootPkg\\":{\\"id\\":\\"/\\",\\"name\\":\\"root-package\\",\\"version\\":\\"1.0.0\\"},\\"_warnings\\":[\\"warn 1\\",\\"warn 2\\",\\"warn\\"],\\"_webpack\\":{\\"logs\\":{}}}
				</script>
				<script>
					window.report = JSON.parse(
						document.getElementById(\\"report\\").innerHTML
					);
				</script>
				<script>
					window.onload = function() {
						var style = document.createElement('style');

						style.innerHTML = '';

						document.head.appendChild(style);

						var select = document.getElementById('log-level-select');

						select.value = 'debug';

						select.onchange = function() {
							switch(select.value) {
								case 'debug':
									style.innerHTML = '';
									break;

								case 'info':
									style.innerHTML = 
										'tr.debug {display: none;}';
									break;

								case 'warn':
									style.innerHTML = 
										'tr.debug {display: none;}' +
										'tr.info {display: none;}';
									break;

								case 'error':
									style.innerHTML = 
										'tr.debug {display: none;}' +
										'tr.info {display: none;} ' +
										'tr.warn {display: none;}';
									break;
							}
						};
					}
				</script>
			</head>
			<body>
				<div id='log-level-selector'>
					Log level filter: 
					<select id='log-level-select'>
						<option>debug</option>
						<option>info</option>
						<option>warn</option>
						<option>error</option>
					</select>
				</div>
				
				<h1>Report of liferay-npm-bundler execution</h1>
				
			<table>
				<tr class=\\"\\"><td>Executed at:</td><td>Thu, 01 Jan 1970 00:00:00 GMT</td></tr>
<tr class=\\"\\"><td>Execution took:</td><td>1s</td></tr>
			</table>
		
				
		<h2>Warnings</h2>
		
		<ul>
			<li>warn 1</li> <li>warn 2</li> <li>warn</li>
		</ul>
	
	
				
		<h2>Project information</h2>
		
			<table>
				<tr class=\\"\\"><th>Name</th>
<th>Version</th></tr>
				<tr class=\\"\\">
				<td>root-package</td>
				<td>1.0.0</td>
			</tr>
			</table>
		
	
				
		<h2>Bundler environment versions</h2>
		
			<table>
				<tr class=\\"\\"><th>Package</th>
<th>Version</th></tr>
				<tr class=\\"\\">
					<td>liferay-npm-bundler</td>
					<td>1.4.2</td>
				</tr>
<tr class=\\"\\">
					<td>liferay-npm-bundler-plugin-inject-angular</td>
					<td>1.4.2</td>
				</tr>
			</table>
		
	
				
		<h2>Details of webpack execution</h2>
		
			<table>
				<tr class=\\"\\"><th>File</th>
<th></th>
<th>Source</th>
<th>Messages</th></tr>
				
			</table>
		
	
			</body>
		</html>
	"
`;

exports[`when describing the run correctly stores executionTime 1`] = `
Report {
  "_executionDate": 1970-01-01T00:00:00.000Z,
  "_executionTime": Array [
    1,
    2,
  ],
  "_rootPkg": undefined,
  "_versionsInfo": Object {},
  "_warnings": Array [],
  "_webpack": Object {
    "logs": Object {},
  },
}
`;

exports[`when describing the run correctly stores root package description 1`] = `
Report {
  "_executionDate": 1970-01-01T00:00:00.000Z,
  "_executionTime": undefined,
  "_rootPkg": Object {
    "id": "/",
    "name": "root-package",
    "version": "1.0.0",
  },
  "_versionsInfo": Object {},
  "_warnings": Array [],
  "_webpack": Object {
    "logs": Object {},
  },
}
`;

exports[`when describing the run correctly stores unique warnings 1`] = `
Report {
  "_executionDate": 1970-01-01T00:00:00.000Z,
  "_executionTime": undefined,
  "_rootPkg": undefined,
  "_versionsInfo": Object {},
  "_warnings": Array [
    "warn",
  ],
  "_webpack": Object {
    "logs": Object {},
  },
}
`;

exports[`when describing the run correctly stores versions info 1`] = `
Report {
  "_executionDate": 1970-01-01T00:00:00.000Z,
  "_executionTime": undefined,
  "_rootPkg": undefined,
  "_versionsInfo": Object {
    "liferay-npm-bundler": Object {
      "path": "path/to/bundler",
      "version": "1.4.2",
    },
    "liferay-npm-bundler-plugin-inject-angular": Object {
      "path": "path/to/plugin",
      "version": "1.4.2",
    },
  },
  "_warnings": Array [],
  "_webpack": Object {
    "logs": Object {},
  },
}
`;

exports[`when describing the run correctly stores warnings 1`] = `
Report {
  "_executionDate": 1970-01-01T00:00:00.000Z,
  "_executionTime": undefined,
  "_rootPkg": undefined,
  "_versionsInfo": Object {},
  "_warnings": Array [
    "warn 1",
    "warn 2",
  ],
  "_webpack": Object {
    "logs": Object {},
  },
}
`;
